sequence_task_1/student/include/impl.h

```diff
@@ -20,31 +21,79 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    using size_type = std::size_t;
+    using const_iterator = typename Container::const_iterator;
+    using value_type = typename Container::value_type;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using difference_type = typename Container::difference_type;
 
     /** @todo Iterators */
+    const_iterator begin() const
+    {
+        return m_tracklist.begin();
+    }
+
+    const_iterator end() const
+    {
+        return m_tracklist.end();
+    }
 
     /** @todo Constructor from any reversible sequence container */
+    template <typename SequenceContainer>
+    StaticPlaylist(SequenceContainer& s)
+        :m_tracklist(s.begin(), s.end()) {}
 
     /** @todo Assignment from any reversible sequence container */
+    template <typename SequenceContainer>
+    StaticPlaylist& operator = (SequenceContainer& s)
+    {
+        m_tracklist.assign(s.begin(), s.end());
+        return *this;
+    }
 
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song& play(Args&&... songData);
+    const Song& play(Args&&... songData)
+    {
+        m_tracklist.insert(m_tracklist.begin(), {Song(songData) ...});
```

Comment:
You don't need to construct a temporary object, if you already have the initialization data, that can be used for in-place construction.